etcdserver/v3_server.go

```diff
@@ -803,3 +805,101 @@ func (s *EtcdServer) AuthInfoFromCtx(ctx context.Context) (*auth.AuthInfo, error
 	return authInfo, nil
 
 }
+
+func (s *EtcdServer) Downgrade(ctx context.Context, r *pb.DowngradeRequest) (*pb.DowngradeResponse, error) {
+	switch r.Action {
+	case pb.DowngradeRequest_VALIDATE:
+		return s.downgradeValidate(ctx, r.Version)
+	case pb.DowngradeRequest_ENABLE:
+		return s.downgradeEnable(ctx, r)
+	case pb.DowngradeRequest_CANCEL:
+		return s.downgradeCancel(ctx)
+	}
+	return nil, nil
+}
+
+func (s *EtcdServer) downgradeValidate(ctx context.Context, v string) (*pb.DowngradeResponse, error) {
+	resp := &pb.DowngradeResponse{}
+	var err error
+
+	targetVersion, err := semver.NewVersion(v)
+	if err != nil {
+		return nil, fmt.Errorf("wrong version format: %v", err)
```

Comment:
Let's define these error strings in https://github.com/etcd-io/etcd/blob/master/etcdserver/errors.go
Then map them to proper gRPC error code in 
https://github.com/etcd-io/etcd/blob/master/etcdserver/api/v3rpc/util.go
and 
https://github.com/etcd-io/etcd/blob/master/etcdserver/api/v3rpc/rpctypes/error.go

Reply:
Added